### PR TITLE
Add assertion failure macros

### PR DESCRIPTION
**Pull request by Anderas**
Add macros and `MXLog` methods to log assertions that "crash" the app in DEBUG mode. This is a useful tool during development: wherease merely logging to console is easy to miss, an assertion forces the developer to debug a given issue whilst the context is still in place.

### CODE REVIEW

File: MatrixSDK/Data/RoomList/CoreData/MXCoreDataRoomListDataFetcher.swift [hunk 1]
```diff
@@ -114,10 +113,8 @@ internal class MXCoreDataRoomListDataFetcher: NSObject, MXRoomListDataFetcher {
         return result
     }
     
-    internal init(session: MXSession?,
-                  fetchOptions: MXRoomListDataFetchOptions,
+    internal init(fetchOptions: MXRoomListDataFetchOptions,
```
Comment (Anderas) [inline MatrixSDK/Data/RoomList/CoreData/MXCoreDataRoomListDataFetcher.swift:116]: Unrelated change, the `session` is never used so can be removed from init parameters

**Review by pixlwave — APPROVED** [review]
LGTM 👍
